app/src/views/GrantRegistryGrantDetail.vue

```diff
@@ -324,10 +322,13 @@ function useGrantDetail() {
   const { signer, provider, userAddress, grantRegistry, isCorrectNetwork } = useWalletStore();
   const route = useRoute();
 
+  // --- show unverified grants if unverified = true
+  const grants = route.query.unverified ? allGrants : approvedGrants;
+
   // --- expose grant data ---
   const grant = computed(() => {
     if (!grants.value) return null; // array type unsupported
-    return grants.value[grantId.value];
+    return grants.value.filter((grant) => grant.id.toString() == route.params.id)[0];
```

Comment:
Why was this changed? The original approach was O(1) and now this is O(n), so it might be slow if there's a lot of grants (though realistically the speed change will probably be unnoticeable)

Reply:
just looked into it, and I believe it's because `grantId.value` is the index value for non-whitelisted grants, but since we're fetching the specific grant from whitelisted grants, there is an index mismatch and it needs to pull the id instead so it's filtering for that

there probably is a more efficient way to handle this though, and this might be related to the other comment about making non-whitelisted grants viewable / unviewable

Reply:
Ah, that makes total sense, thanks!